### PR TITLE
gcp-vpc-move-route.in: python implementation of gcp-vpc-move-ip.in

### PR DESCRIPTION
**Pull request by helen-fornazier**
gcloud api is not reliable and it is slow, add a python version of
gcp-vpc-move-ip.in

gcp-vpc-move-ip.in: add deprecation message

### CODE REVIEW

File: heartbeat/gcp-vpc-move-ip.in [hunk 1]
```diff
@@ -348,6 +348,8 @@ ip_and_route_stop() {
 #
 ###############################################################################
 
+ocf_log warn "gcp-vpc-move-ip is deprecated, prefer to use gcp-vpc-move-route instead"
```
Comment (oalbrigt) [inline heartbeat/gcp-vpc-move-ip.in]: Perfect.

File: heartbeat/gcp-vpc-move-route.in [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+route
+2) compute.networks.updatePolicy on the network (to add a new route)
+3) compute.networks.get on the network (to check the VPC network existence)
+4) compute.routes.list on the project (to check conflicting routes)
+</longdesc>
+<shortdesc lang="en">Move IP within a GCP VPC</shortdesc>
+
+<parameters>
+
+<parameter name="stackdriver_logging" unique="0" required="0">
```
Comment (oalbrigt) [inline heartbeat/gcp-vpc-move-route.in]: I would move this to the bottom of the parameter list, so we got the general parameters first.

File: heartbeat/gcp-vpc-move-route.in [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+<shortdesc lang="en">Floating IP</shortdesc>
+<content type="string" />
+</parameter>
+
+<parameter name="vpc_network" required="1">
+<longdesc lang="en">
+Name of the VPC network
+</longdesc>
+<shortdesc lang="en">VPC network</shortdesc>
+<content type="string" default="${OCF_RESKEY_vpc_network_default}" />
```
Comment (oalbrigt) [inline heartbeat/gcp-vpc-move-route.in]: You should clear out rhese ${...} in default, (seems like you got a few of them from copy-paste), as they're used to get the default-value in the shell script agents.

File: heartbeat/gcp-vpc-move-route.in [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+
+  configure_logs(ctx)
+  if 'start' in sys.argv[1]:
+    ip_and_route_start(ctx)
+  elif 'stop' in sys.argv[1]:
+    ip_and_route_stop(ctx)
+  elif 'status' in sys.argv[1] or 'monitor' in sys.argv[1]:
+    sys.exit(ip_monitor(ctx))
+  else:
+    usage = 'usage: $0 {start|stop|monitor|status|meta-data|validate-all}'
```
Comment (oalbrigt) [inline heartbeat/gcp-vpc-move-route.in]: $0 is os.path.basename(sys.argv[0]) in Python.

File: heartbeat/gcp-vpc-move-route.in (hunk 3 above)
Comment (helen-fornazier) [reply heartbeat/gcp-vpc-move-route.in]: ops, sorry about that, fixed

File: heartbeat/gcp-vpc-move-route.in (hunk 4 above)
Comment (helen-fornazier) [reply heartbeat/gcp-vpc-move-route.in]: fixed

File: heartbeat/gcp-vpc-move-route.in [hunk 5]
```diff
@@ -0,0 +1,442 @@
...
+
+<actions>
+<action name="start" timeout="180s" />
+<action name="stop" timeout="180s" />
+<action name="monitor" depth="0" timeout="30s" interval="60s" />
+<action name="validate-all" timeout="5s" />
+<action name="meta-data" timeout="5s" />
+</actions>
+</resource-agent>
+''' % os.environ['__SCRIPT_NAME']
```
Comment (oalbrigt) [inline heartbeat/gcp-vpc-move-route.in]: It fails due to this, so I guess you can change it to "os.path.basename(sys.argv[0])"

File: heartbeat/gcp-vpc-move-route.in [hunk 6]
```diff
@@ -0,0 +1,442 @@
...
+    sys.exit(OCF_ERR_CONFIGURED)
+
+  ctx.instance_url = '%s/projects/%s/zones/%s/instances/%s' % (
+      GCP_API_URL_PREFIX, ctx.project, ctx.zone, ctx.instance)
+  ctx.vpc_network = os.environ.get('OCF_RESKEY_vpc_network', 'default')
+  ctx.vpc_network_url = '%s/projects/%s/global/networks/%s' % (
+      GCP_API_URL_PREFIX, ctx.project, ctx.vpc_network)
+  ctx.interface = os.environ.get('OCF_RESKEY_interface', 'eth0')
+  ctx.route_name = os.environ.get(
+      'OCF_RESKEY_route_name', 'ra-%s' % os.environ['__SCRIPT_NAME'])
```
Comment (helen-fornazier) [inline heartbeat/gcp-vpc-move-route.in]: should I change here to "os.path.basename(sys.argv[0])" too?

Comment (oalbrigt) [reply heartbeat/gcp-vpc-move-route.in]: You might as well for consistency.

File: heartbeat/gcp-vpc-move-route.in (hunk 5 above)
Comment (helen-fornazier) [reply heartbeat/gcp-vpc-move-route.in]: done

File: heartbeat/gcp-vpc-move-route.in (hunk 6 above)
Comment (helen-fornazier) [reply heartbeat/gcp-vpc-move-route.in]: done

**Comment by oalbrigt** [comment]
Thanks.

**Pull request merged**